adrs/0049-prebuilt-rubies-at-runtime.md

```diff
@@ -3,43 +3,78 @@
 
 This is in contrast to other setup actions like setup-node where the cache is an optimization but supports pulling any version by a specified semver at runtime.
 
+Ruby officially only supports source code distribution.  There are no "official" pre built distributions since there are system and environmental dependencies.
+
 ## Goals
 
 - Offer as many versions as possible by platforms
 - Offer popular Ruby variants like JRuby
 - Consistent with other setup-xxx actions patterns and Actions workflows.
+- Ideally works across all supported scenarios including self-hosted runners.
+- Ideally works on all the supported runner platforms (windows, mac-os, all linux variants)
+- Ideally works with actions container scenarios
 
-## Decisions
+## Options
 
-### Prebuilt Rubies
+### Download Prebuilt Rubies on Misses
 
 Other tools like offer a [distribution](https://nodejs.org/dist/) which offers a [queryable endpoint](https://nodejs.org/dist/index.json) which allows the desired version [by a version spec which is semver](https://github.com/actions/setup-node#setup-node).
 
 Other CI services like Travis support pulling a wide variety of [ruby versions offered here](http://rubies.travis-ci.org/).
 
-Requirements:
-- Offer prebuilt Ruby, JRuby, TruffleRuby versions 
-- Setup-Ruby exposes `ruby-version` which accepts a semver
-- Rubies are discoverable and queryable by the Setup-Ruby action.
-- Initially offer for all platforms offered by the hosted Actions VMs (Ubuntu 18, Mac 10.15 and Windows 2019)
-
 NOTE: The prebuilt rubies offered will only be supported for use by GitHub Actions.
 
 The scripts and tooling to build the Rubies will be at `actions/build-ruby` and it will be consumed by the existing action `actions/setup-ruby`.  Since the capabilities are additive, there's no need to create a `v2`.  There is no compat break.
 
-A specific version of a Ruby will be offered as an individual semi-immutable build-ruby repo release similar to how the [actions runner exposes versions](https://github.com/actions/runner/releases/tag/v2.164.0) where each version offers n platforms.  This is also similar to [Travis individual versions](http://rubies.travis-ci.org/ubuntu/18.04/s390x/ruby-2.6.5).
+A specific version of a Ruby will be offered as an individual build-ruby repo release similar to how the [actions runner exposes versions](https://github.com/actions/runner/releases/tag/v2.164.0) where each version offers n platforms.  This is also similar to [Travis individual versions](http://rubies.travis-ci.org/ubuntu/18.04/s390x/ruby-2.6.5).
+
+**Pros**
+
+On a toolkit cache miss, the first build is slower but not that slow since it's pre-built.
+
+**Cons**
+
+Will not work with self-hosted scenarios
+Will not work across all platforms the runner supports
```

Comment:
Why not?

Reply:
It effectively won't work against all linux variants (rh, ubuntu, arch, arm etc.) times all versions  (18.04, 18.10 etc)  which our runner currently supports.  From discussions, this is why Ruby only officially supports building from source - the environmental and dependency differences.  That is unless you want to build an impossible matrix of platform, all versions, arch etc. prebuilt.

Reply:
Ah right, I understood that as "all the platforms GitHub-hosted runners support".
So it's related to the point above about self-hosting and more platforms.